R/Previewer.R

```diff
@@ -124,122 +124,118 @@ reporter_previewer_srv <- function(id,
     )
     report_load_srv("load", reporter = reporter)
     reset_report_button_srv("reset", reporter = reporter)
+    reporter_previewer_cards_srv("cards", reporter)
+  })
+}
 
-    current_cards <- shiny::reactiveVal()
-    insert_cards <- shiny::reactiveVal()
-    remove_cards <- shiny::reactiveVal()
+reporter_previewer_cards_ui <- function(id) {
+  ns <- shiny::NS(id)
+  shiny::tags$div(
+    id = "reporter_previewer",
+    tags$div(
+      id = ns("empty_reporters"),
+      tags$h4(
+        class = "text-muted",
+        shiny::icon("circle-info"),
+        "No reports have been added yet."
+      )
+    ),
+    bslib::accordion(id = ns("reporter_cards"), open = FALSE)
+  )
+}
+
+reporter_previewer_cards_srv <- function(id, reporter) {
+  moduleServer(id, function(input, output, session) {
+    current_cards_rvs <- shiny::reactiveValues()
+    current_hash_rvs <- shiny::reactiveValues()
+
+    insert_queue_rv <- shiny::reactiveVal()
+    remove_queue_rv <- shiny::reactiveVal()
     shiny::observeEvent(reporter$get_reactive_add_card(), {
-      reporter_ids <- vapply(reporter$get_cards(), attr, character(1L), which = "id")
-      current_ids <- vapply(current_cards(), attr, character(1L), which = "id")
-
-      to_add <- reporter$get_cards()[!reporter_ids %in% current_ids]
-      to_remove <- current_cards()[!current_ids %in% reporter_ids]
-      if (length(to_add)) insert_cards(to_add)
-      if (length(to_remove)) remove_cards(to_remove)
-      current_cards(reporter$get_cards())
+      all_cards <- reporter$get_cards()
+
+      reporter_ids <- vapply(all_cards, metadata, character(1L), USE.NAMES = FALSE, which = "id")
```

Comment:
`"id"` doesn't seem to be needed.
```suggestion
      reporter_ids <- names(all_cards)
```